frontend/src/layout/RTCLayout.tsx

```diff
@@ -0,0 +1,60 @@
+import { useCallback, useEffect } from "react";
+import { Outlet } from "react-router-dom";
+import { useRecoilValueLoadable, useSetRecoilState } from "recoil";
+import { ensembleAtom, userListAtom } from "../recoil/ensemble";
+import { socketAtom } from "../recoil/socket";
+import { Ensemble, EnsembleObject } from "common/dist";
+
+/**
+ * This is a separate layout for Real Time Collaboration (RTC)
+ *
+ * When navigating to the Ensemble view, this component gets loaded.
+ * When this happens, we communicate with the server to set up the necessary parts
+ * for RTC, using userListAtom and ensembleAtom, before it gets used.
+ *
+ * We do this to be ready to "catch" the information given to us by the server when we join an Ensemble.
+ * Otherwise, the information is sent too quickly for us to receive, store, and utilize the data.
+ *
+ * @returns
```

Comment:
What does it return? /s

I'd probably remove the @returns. This is a good docstring